dsgrid/config/time_dimension_base_config.py

```diff
@@ -0,0 +1,77 @@
+import abc
+
+from .dimension_config import DimensionBaseConfigWithoutFiles
+
+
+class TimeDimensionBaseConfig(DimensionBaseConfigWithoutFiles, abc.ABC):
+    """Base class for all time dimension configs"""
+
+    @abc.abstractmethod
+    def check_dataset_time_consistency(self, load_data_df):
+        """Check consistency of the load data with the time dimension.
+
+        Parameters
+        ----------
+        load_data_df : pyspark.sql.DataFrame
+
+        Raises
+        ------
+        DSGInvalidDataset
+            Raised if the dataset is inconsistent with the time dimension.
+
+        """
+
+    @abc.abstractmethod
+    def convert_dataframe(self, df):
+        """Convert a dataframe to standard timestamps.
```

Comment:
What are "standard" timestamps?

Reply:
That is a weak description. I'll change it to something like python datetime timestamps.